src/Components/OATGraphViewer/Internal/OATGraphCustomEdge.tsx

```diff
@@ -49,35 +49,127 @@ const OATGraphCustomEdge: React.FC<IOATGraphCustomEdgeProps> = ({
     } = useContext(ElementsContext);
     const graphViewerStyles = getGraphViewerStyles();
     const theme = useTheme();
+    sourceY = sourceY - 6;
 
     const element = elements.find((x) => x.id === id);
+    let inheritancePolygon = '';
```

Comment:
This is a significant amount of logic to have in the main render flow. Can we please break it up into well named methods? We should also consider using `useMemo` so we can ensure it doesn't run unless something actually changes. Also, a few comments might be good to explain the underlying algorithm that has been implemented.

Reply:
Thanks for turning this into a `useMemo`. Can we please also break it up into well named methods? Rather than one big long one?

Reply:
Created 2 new methods getSourceComponents and getTargetComponents let me know if you have any more questions